backtracking/rat_in_maze.py

```diff
@@ -1,81 +1,155 @@
 from __future__ import annotations
 
 
-def solve_maze(maze: list[list[int]]) -> bool:
+def solve_maze(
+    maze: list[list[int]],
+    source_row: int,
+    source_column: int,
+    destination_row: int,
+    destination_column: int,
+) -> list[list[int]]:
     """
     This method solves the "rat in maze" problem.
-    In this problem we have some n by n matrix, a start point and an end point.
-    We want to go from the start to the end. In this matrix zeroes represent walls
-    and ones paths we can use.
     Parameters :
-        maze(2D matrix) : maze
+        - maze(2D matrix) : maze
+        - source_row (int): The row index of the starting point.
+        - source_column (int): The column index of the starting point.
+        - destination_row (int): The row index of the destination point.
+        - destination_column (int): The column index of the destination point.
```

Comment:
Do not repeat the datatypes which are checked by mypy in the function signature above.  Readers will get confused if one changes and the other does not.
```suggestion
        - maze: A two dimensional matrix of zeros and ones.
        - source_row: The row index of the starting point.
        - source_column: The column index of the starting point.
        - destination_row: The row index of the destination point.
        - destination_column: The column index of the destination point.
```